internal/grpc/grpclog/grpclog.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+// grpclog provides a logging utilities for gRPC.
+package grpclog
+
+import (
+	"context"
+	"sync"
+
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/peer"
+)
+
+// PeerAddrFromContext returns a peer's address from context, or "unknown" if not found.
+func PeerAddrFromContext(ctx context.Context) string {
+	p, ok := peer.FromContext(ctx)
+	if !ok {
+		return "unknown"
+	}
+	return p.Addr.String()
+}
+
+// LogStateChanges logs the state changes of a gRPC connection.
+func LogStateChanges(ctx context.Context, conn getStater, log debugLog, wg *sync.WaitGroup, isReadyCallback func()) {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		state := conn.GetState()
+		log.Debugf("Connection state started as %s", state)
+		for ; state != connectivity.Ready && conn.WaitForStateChange(ctx, state); state = conn.GetState() {
```

Comment:
```suggestion
		for ; conn.WaitForStateChange(ctx, state); state = conn.GetState() {
```
If I remove this part, tests still pass. We should add a test case for the break condition.